### DISCUSSION TITLE
PlayAssetDelivery ANE Examples?

### DISCUSSION THREAD

**Discussion by SesameWorkshopApps** [airsdk/Adobe-Runtime-Support#2761] [Guides & Tutorials]
Greetings,, I'm looking for some sample implementations of the PlayAssetDelivery ANE. I'm having a bit of difficulty enabling in my project. For instance, I keep getting an error trying to import the library:

    [mxmlc] Warning: Definition com.harman.extension.PlayAssetDelivery could not be found.
    [mxmlc] 	import com.harman.extension.PlayAssetDelivery;

If I had some reference as to how it's properly imported, I'd have an easier time troubleshooting my own project.

Thanks,

Jonathan

**Comment by marchbold** [comment]
Hi,

Firstly check you have correctly added the extensions to your extension list in your app descriptor:

```xml
<extensions>
    <extensionID>com.harman.PlayAssetDelivery</extensionID>
    <extensionID>com.google.android.play</extensionID>
...
``` 

and that you have added it to your application dependencies.  

Then the import is :

```actionscript
import com.harman.extension.PlayAssetDelivery;
```

When packaging with mxmlc you need to ensure you have linked the ANEs as an `external-library-path` (if using a command line implementation we normally just rename / copy the ANEs to SWCs for this, though most IDEs should do this for you).

**Reply by SesameWorkshopApps** [reply]
Greetings @marchbold , thank you for the response. In my case, the issue was that I re-packed the SWC. Replacing it with the ANE got me farther.
Now I simply get this error:

    [mxmlc] Error: Call to a possibly undefined method openInstallTimeAsset.
    [mxmlc] 				var asset:AssetFile = openInstallTimeAsset( src );
    [mxmlc] 				                      ^

**Reply by marchbold** [reply]
Are you trying to call the function on the `PlayAssetDelivery` instance?

```as3
var assets:PlayAssetDelivery = new PlayAssetDelivery();

if (assets.initAssetDelivery())
{
    // PlayAssetDelivery initialised
    var asset:AssetFile = assets.openInstallTimeAsset( "path/example.txt" );

}
